Sources/Orbit/Components/SegmentedSwitch.swift

```diff
@@ -0,0 +1,157 @@
+
+    @Binding private var selection: Value?
+    let label: String
+    let _message: Message?
+    let firstOption: Value
+    let secondOption: Value
+
+    let separatorWidth: CGFloat = .xxxSmall
+    let borderWidth: CGFloat = .xxxSmall
+
+    public var body: some View {
+        FieldWrapper(
+            label,
+            message: message,
+            content: {
+                HStack(spacing: 0) {
+                    segment(title: firstOption)
+                    separator
+                    segment(title: secondOption)
+                }
+                .background(selectionBackground)
+                .background(roundedBackground)
+                .fixedSize(horizontal: false, vertical: true)
+            }
+        )
+    }
+
+    @ViewBuilder func segment(title: Value) -> some View {
+        Text(title.description)
+            .lineLimit(1)
+            .padding(.vertical, .xSmall)
+            .frame(maxWidth: .infinity)
+            .contentShape(Rectangle())
+            .onTapGesture {
+                selection = title
+            }
+    }
+
+    @ViewBuilder var separator: some View {
+        Rectangle()
+            .fill(borderColor)
+            .frame(width: separatorWidth)
+    }
+
+    @ViewBuilder var selectionBackground: some View {
+        GeometryReader { reader in
+            if selection != nil {
+                Color.whiteDarker
+                    .frame(width: (reader.size.width - 2 * borderWidth) / 2)
+                    .clipShape(RoundedRectangle(cornerRadius: BorderRadius.default))
+                    .padding(borderWidth)
+                    .elevation(.level1)
+                    .offset(computeSelectionOffset(in: reader))
+                    .animation(.easeOut(duration: 0.2), value: selection)
+            }
+        }
+    }
+
+    @ViewBuilder var roundedBackground: some View {
+        RoundedRectangle(cornerRadius: BorderRadius.default)
+            .fill(backgroundColor)
+            .overlay(
+                RoundedRectangle(cornerRadius: BorderRadius.default)
+                    .stroke(borderColor, lineWidth: borderWidth)
+            )
+    }
+
+    private var message: Message? {
+        if case .error = _message, selection != nil {
+            return nil
+        }
+
+        return _message
+    }
+
+    private var backgroundColor: Color {
+        selection != nil ? .cloudNormal : .whiteDarker
+    }
+
+    private var borderColor: Color {
+        guard selection == nil else { return .clear }
+
+        switch message {
+            case .error:
+                return .redNormal
+            default:
+                return .cloudNormal
+        }
+    }
+
+    private func computeSelectionOffset(in reader: GeometryProxy) -> CGSize {
+        .init(width: selection == firstOption ? 0 : reader.size.width / 2 - borderWidth, height: 0)
+    }
+
+    public init(
+        label: String,
+        message: Message? = nil,
+        firstOption: Value,
```

Comment:
We should follow existing APIs:
- `_ label: String,`
- `message` should go last, to follow the left-right, top-bottom order of components

```

    public init(
        _ label: String,
        firstOption: Value,
        secondOption: Value,
        message: Message? = nil,
        selectedOption: Binding<Value?>
    )
```

Reply:
The `_ label` is still missing.

Reply:
updated